### ISSUE TITLE
Allow adding an observation from anywhere the NavBar is visible

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#462]
Currently you get a navigation error any time you do this from outside My Observations. If the NavBar is visible, tapping all the buttons in the AddEvidenceModal should work.

This will probably be pretty disruptive since it will involve rethinking how we've organized our navigators (it's not clear to me why we have multiple in the first place).

**Comment by Chrischuck** [comment]
I can take a look at this later today. Will probably get a proposal first in this issue thread before doing any real coding since, like you said, looks pretty involved.

**Comment by Chrischuck** [comment]
Briefly looking, I think it's reasonable and a pretty common pattern to have two navigators (one drawer, one stack)

I think one possible solution might look like the following in a sub stack (mainStackNavigation, projectStackNavigation, etc.)
```
import SharedStackScreens from './SharedStackScreens'
    <Stack.Navigator screenOptions={showHeader}>
      <Stack.Screen
        name="StandardCamera"
        component={StandardCameraWithPermission}
        options={{ ...hideHeader, orientation: isTablet ? "all" : "portrait" }}
      />
    {SharedStackScreens}
    ...
```

I haven't written any code yet so I'm not sure if there will be any mismatched state issues...

**Comment by Chrischuck** [comment]
Above did not work...Looks like the lowest barrier to entry is simply specifying which stack you want to navigate to. Like: `navigation.navigate( "MainStack", { screen, ...params } );` Though I'm trying to find a clean way to do this

**Comment by Chrischuck** [comment]
Looks like an issue many have had with react navigation. See [here](https://github.com/react-navigation/react-navigation/issues/6434)

implemented a solution like this [one](https://github.com/react-navigation/react-navigation/issues/6434#issuecomment-604523082)

It seems to work fairly well. Able to go from `ProjectDetails` to `ObsEdit` and back to the project you were viewing (if you discard the observation you are editing)

**Comment by albullington** [comment]
Closed via #507
